.github/workflows/.lint.yaml

```
@@ -5,56 +5,228 @@ permissions:
+  pre-commit:
+    name: pre-commit
     runs-on: ubuntu-latest
-    permissions:
-      contents: write
     environment:
       name: code_quality
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@93cb6efe18208431cddfb8368fd83d5badbf9bfd # v5
+      - uses: actions/checkout@v5
+      - uses: actions/setup-python@v3
```

actionlint (1.7.9)

the runner of "actions/setup-python@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)